client-device-connection-infinispan/src/main/java/org/eclipse/hono/deviceconnection/infinispan/client/CacheBasedDeviceConnectionClient.java

```diff
@@ -81,21 +80,25 @@ public CacheBasedDeviceConnectionClient(final DeviceConnectionInfo cache, final
 
     /**
      * {@inheritDoc}
+     * <p>
+     * Simply delegates to {@link #registerCommandConsumer(String, String, String, Duration, SpanContext)}.
      */
     @Override
-    public Future<JsonObject> getLastKnownGatewayForDevice(
+    public Future<Void> setCommandHandlingAdapterInstance(
             final String tenantId,
             final String deviceId,
+            final String adapterInstanceId,
+            final Duration lifespan,
             final SpanContext context) {
 
-        final Span span = newSpan(context, SPAN_NAME_GET_LAST_GATEWAY);
-        TracingHelper.setDeviceTags(span, tenantId, deviceId);
-
-        return finishSpan(connectionInfoCache.getLastKnownGatewayForDevice(tenantId, deviceId, span), span);
+        return registerCommandConsumer(tenantId, deviceId, adapterInstanceId, lifespan, context);
     }
 
+    /**
+     * {@inheritDoc}
+     */
     @Override
-    public Future<Void> setCommandHandlingAdapterInstance(
+    public Future<Void> registerCommandConsumer(
```

Comment:
FMPOV, an `UnsupportedOperationException` should be thrown here. The `registerCommandConsumer` method should not be invoked on the DeviceConnectionClient.

Same goes for the `unregisterCommandConsumer` method.